### PR TITLE
IA-1896 Add contributing guide

### PR DESCRIPTION
**Pull request by gpcarr**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributing
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:1]: can you link this in the main `README.md` as well

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+`terra-docker` consists of docker images that are compatible with the notebooks service in Terra called Leonardo. We follow the naming convention of `terra-[application]-[image_name]` (for example `terra-jupyter-aou`).
+
+## Opening a PR
+
+We do our work on a feature branch based on master. When you open a pull request, add the JIRA issue number to the PR title. This will make a reference from JIRA to the GitHub issue. Briefly describe your changes in the description section.
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:7]: we expect external contributors too other than broad. Maybe add something like indicating no need to care about JIRA if you're from outside broad?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+This is also a good opportunity to verify that all the appropriate image versions were bumped. It is important to keep in mind which images extend from where ([Image Dependecies](https://github.com/DataBiosphere/terra-docker#image-dependencies)). Make sure to bump the version of the image you are changing and update its reference in all images that extend from it.
+
+## Merging your PR
+
+Make sure to get two thumbs before merging your PR. We often reach out to the AOU or Bioconductor team to also review changes we are making to their images.
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:13]: maybe mention bioconductor team for r image as well

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Once you merge your PR, this will automatically kick off a Jenkins job (given title "terra-docker image hash update") to create a PR in Leonardo that bumps the image versions in [reference.conf](https://github.com/DataBiosphere/leonardo/blob/develop/automation/src/test/resources/reference.conf) under automation. This allows for our automation tests in Leonardo to run against your new docker image changes. There will be a notification in slack once this job finishes running with a link to the automatically created Leonardo PR.
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:15]: mention `dsp-callisto-internal` channel?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+The terra-docker image hash update [job](https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/)  mentioned above determines which images (including derived images) have been changed and builds each image with the appropriate tag.
+
+## Merging the terra-docker image hash update job PR
+
+Once all automation tests (including coverage) have passed, you can merge this PR. It's not absolutely necessary to get a review on this PR since your `terra-docker` PR changes were reviewed already.
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:21]: I would still get a thumb from one other team member

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Merging this PR will automatically kick off another Jenkins [job](https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-dataproc-image/) (given title "Automated custom-image hash update"). This job generates a file called `terra-docker-versions-new.json` which we will use later.
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:23]: can you mention somewhere we should add the jira number to these auto generated PRs just so it's easier to know what these PRs are for

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Merging the terra-docker custom image hash update job PR
+
+Merging this PR means that the new terra-docker versions will be available in Leonardo. If there are any Leo changes that go along with your `terra-docker` changes, this is a good place to push them to.
+
+## Terra UI representation
```
Comment (Qi77Qi) [inline CONTRIBUTING.md:29]: can we remove https://github.com/databiosphere/terra-docker#terra-ui-representation now

File: CONTRIBUTING.md (hunk 1 above)
Comment (gpcarr) [reply CONTRIBUTING.md:1]: will do

File: CONTRIBUTING.md (hunk 2 above)
Comment (gpcarr) [reply CONTRIBUTING.md:7]: yeah I can add that. I can also maybe add a sentence saying "there's still more to come on external contributions"

File: CONTRIBUTING.md (hunk 3 above)
Comment (gpcarr) [reply CONTRIBUTING.md:13]: good point

File: CONTRIBUTING.md (hunk 4 above)
Comment (gpcarr) [reply CONTRIBUTING.md:15]: will do

File: CONTRIBUTING.md (hunk 5 above)
Comment (gpcarr) [reply CONTRIBUTING.md:21]: okay i'll update. i'll add the same comment for the second autocreated PR

File: CONTRIBUTING.md (hunk 7 above)
Comment (gpcarr) [reply CONTRIBUTING.md:29]: yep, i can do that

**Comment by gpcarr** [comment]
> looks great....a few minor comments

addressed all comments :)

**Pull request merged**